logstash-core/lib/logstash/util/time_value.rb

```diff
@@ -71,8 +72,12 @@ def to_nanos
         end
       end
 
+      def to_seconds
+        self.to_nanos / 1_000_000_000.0
+      end
+
       def ==(other)
-        self.duration == other.duration and self.time_unit == other.time_unit
+        (self.duration == other.duration && self.time_unit == other.time_unit) || self.to_nanos == other.to_nanos
```

Comment:
review note: I consider two `TimeValue` objects to be equivalent if they resolve to the same number of nanoseconds. This marginally changes the semantics of this method, which previously said "60s" and "1m" were _not_ equal. But the equality was never used in practice because this was effectively an intermediate object that always was converted to nanos.

This equivalence check will short-circuit and perform no math if the two objects are _equal_, and will fall back to the math only necessary.